integration-tests/tests/src/performance-tests/property-reads.ts

```diff
@@ -16,53 +16,59 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
-import Realm, { ObjectSchema, PropertySchema } from "realm";
+import { expect } from "chai";
+import Realm, { BSON, ObjectSchema, PropertySchema, PropertySchemaShorthand } from "realm";
 
 import { describePerformance } from "../utils/benchmark";
 
 type Value = ((realm: Realm) => unknown) | unknown;
 
-function getTypeName(type: Realm.PropertySchemaShorthand | Realm.PropertySchema) {
-  if (typeof type === "object") {
-    const prefix = type.optional ? "optional " : "";
-    if (type.objectType) {
-      return prefix + `${type.type}<${type.objectType}>`;
-    } else {
-      return prefix + type.type;
-    }
-  } else {
-    return type;
+const COLLECTION_MARKERS: Readonly<Record<string, string>> = {
+  list: "[]",
+  dictionary: "{}",
+  set: "<>",
+};
+
+/**
+ * Get a representative consistent name of the type depending on the schema.
+ *
+ * @example
+ * "int?[]"                                            -> "int?[]"
+ * { type: "list", objectType: "int", optional: true } -> "int?[]"
+ */
+function getTypeDisplayName(schema: PropertySchemaShorthand | PropertySchema) {
```

Comment:
Renaming from `getTypeName` to disambiguate it from the various usages of `type` we have.